pkg/slack/filter.go

```diff
@@ -0,0 +1,84 @@
+package slack
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/oursky/github-actions-manager/pkg/github/jobs"
+	"k8s.io/utils/strings/slices"
+)
+
+type messageFilterRule struct {
+	conclusions []string
+	// branches    []string
+	workflows []string
+}
+type MessageFilter struct {
+	whitelists []messageFilterRule
+}
+
+func (rule messageFilterRule) Pass(run *jobs.WorkflowRun) bool {
+	if len(rule.conclusions) > 0 && !slices.Contains(rule.conclusions, run.Conclusion) {
+		return false
+	}
+	if len(rule.workflows) > 0 && !slices.Contains(rule.workflows, run.Name) {
+		return false
+	}
+	return true
+}
+
+func (mf MessageFilter) Any(run *jobs.WorkflowRun) bool {
+	if len(mf.whitelists) == 0 {
+		return true
+	}
+	for _, rule := range mf.whitelists {
+		if rule.Pass(run) {
+			return true
+		}
+	}
+	return false
+}
+
+func (rule *messageFilterRule) setConclusions(conclusions []string) error {
+	conclusionsEnum := []string{"action_required", "cancelled", "failure", "neutral", "success", "skipped", "stale", "timed_out"}
```

Comment:
In golang, we will write enum like this
```go
type Conclusion string

const (
  ConclusionActionRequred Conclusion = "action_requred"
  ConclusionCancelled Conclusion = "cancelled"
  ....
)

```

Then, change the signature of this method to `SetConclusions(conclusions []Conclusion) error`

Reply:
This line is directly taken from the original code, so don't shoot the messenger. :doglaugh: Ok, I'll clean this up.